### PR TITLE
Docs: note about snapshot version compatibility

### LINKED ISSUE #20866: docs: snapshot restore versions
The docs page for all version of snapshot / restore should explain the limitations in how they can be restored in the future.

> You can restore indices that were created in the previous major version only.  Indices that were created before the previous major version (even if they were snapshotted in the previous major version) cannot be restored.

> Create an index in 2.x, snapshot it, restore in 5.x [supported]
> Create an index in 1.x, snapshot it, restore in 2.x [supported]
> Create an index in 1.x, snapshot it in 2.x, restore in 5.x [ NOT supported]
> Create an index in 1.x, reindex it in 2.x, snapshot it in 2.x, restore in 5.x [supported]

### PR DESCRIPTION
**Pull request by nik9000**
It is important that folks understand that snapshot/restore isn't
for archiving. It is appropriate for backup and disaster recovery
but not for archival over long periods of time because of version
incompatibility.

Closes #20866

### CODE REVIEW

**Comment by nik9000** [comment]
@jpcarey, can you have a look at this?

**Review by jpountz — APPROVED** [review]
LGTM

**Comment by jpcarey** [comment]
LGTM

**Comment by nik9000** [comment]
Thanks for reviewing! I'll probably wait until the morning to merge and backport. I'll have to drop the link to reindex from remote if you go before 5.0.0 and just mention that it is a 5.0 feature. Anyway, I don't want to merge tonight and make a mistake and leave the docs build broken all night.

**Comment by nik9000** [comment]
OK! Here goes backporting!

**Comment by nik9000** [comment]
Master: 3bba7dbe07b007804a47cf2460ece5be9979f181
5.x: 4c2ba9c1180fe9a986a42fa874338246d7537a7f
5.0: 443226a00a2fd5e6cdd39ecd62fd9ab5fa6bca8d
2.4: aea9ba6ea7d23f7cfba85f3fd117fc58bd8e16c2
1.7: 314c2a0c5fdd919a964baf332919bccb830b4f4c

**Pull request merged**
